crates/nu-cli/src/commands/documentation.rs

```diff
@@ -0,0 +1,291 @@
+use crate::commands::WholeStreamCommand;
+
+use crate::prelude::*;
+use nu_protocol::{NamedType, PositionalType, Signature, UntaggedValue, Value};
+
+use indexmap::IndexMap;
+
+use std::collections::HashMap;
+
+const COMMANDS_DOCS_DIR: &str = "docs/commands";
+
+pub struct DocumentationConfig {
+    no_subcommands: bool,
+    no_colour: bool,
+}
+
+impl Default for DocumentationConfig {
+    fn default() -> Self {
+        DocumentationConfig {
+            no_subcommands: false,
+            no_colour: false,
+        }
+    }
+}
+
+fn generate_doc(registry: &CommandRegistry, name: &str) -> IndexMap<String, Value> {
+    let mut row_entries = IndexMap::new();
+    let command = registry
+        .get_command(name)
+        .unwrap_or_else(|| panic!("Expected command '{}' from names to be in registry", name));
+    row_entries.insert(
```

Comment:
if you're going to unwrap_or_else into a panic, you could use `.expect(` here

Reply:
I think I did originally, but `clippy` gave me a linter warning for using `.expect` on a function call or something

Reply:
https://rust-lang.github.io/rust-clippy/master/#expect_fun_call